enhancements/update/versioning-cincinnati-api.md

```diff
@@ -0,0 +1,122 @@
+---
+title: versioning-cincinnati-api
+authors:
+- "@pratikmahajan"
+  reviewers:
+- @openshift/openshift-team-cincinnati-maintainers
+- @openshift/openshift-team-cincinnati
+- @openshift/team-cincinnati-operator
+  approvers:
+- @crawford
+- @LalatenduMohanty
+- @sdodson
+  creation-date: 2021-08-20
+  last-updated: 2021-08-20
+  status: implementable
+  see-also:
+-
+replaces: []
+superseded-by: []
+---
+
+# Versioning cincinnati api and json schema
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+As of today, we declare the schema version in Cincinnati Graph URI. We currently, do not embed the version information
+in json schema. The enhancement will embed the version information in the json schema. This has the benefit of being
+easy to implement and to pass around if we ship Cincinnati JSON over non-HTTP channels. The enhancement will also begin
+versioning Cincinnati payloads by media type, for example `application/vnd.redhat.cincinnati.v1+json`.
+The move to supplying requested version in media type field will require us to move towards a generic un-versioned Graph
+URI, with the move, we'll still support old URI for foreseeable future.
+
+## Motivation
+
+### Goals
+1. Embed version information in json schema.
+2. Implement media type versioning for clients requesting a specific schema version of Cincinnati
+3. Move to a generic un-versioned URI for Cincinnati
+4. Cincinnati continues to support older clients after adding support for new clients for foreseeable future.
+5. Add metrics to identify hits on URIs and versions.
+
+### Non-Goals
+
+## Proposal
+
+### Implementation
+
+#### Version information in json schema
+Embed the version information in json schema. The updated json schema should look like:
+```json
+{
+    "version": 1,
+    "nodes": [],
+    "edges": []
+}
+```
+
+#### Media Type Versioning
+Currently, the versioning is done by using the versioned URI and does not support media type versioning. The user should
+be able to request a specific version of cincinnati graph by supplying the version in media type field of the request.
+for example `application/vnd.redhat.cincinnati.v1+json`. Here are some of the
+[examples of media type versioning](https://github.com/opencontainers/image-spec/blob/v1.0.1/media-types.md).  
```

Comment:
I think you are trying to say that we will support the `OCI Image Media Types` as media types is more generic than that https://datatracker.ietf.org/doc/html/rfc6838

Reply:
Media types are generic, and RFC 6838 mentions versions in passing a few times.  But many generic media types are not versioned (e.g. image/jpg).  The OCI image types, on the other hand, include some explicit examples of media types being used for versioning.